evaluation.md

```diff
@@ -0,0 +1,413 @@
+
+This notebook either needs to be executed in the [TAAR](http://github.com/mozilla/taar) repository or somewhere where TAAR is in the Python path, because some TAAR recommenders are loaded in.
+
+## Retrieving the relevant variables from the longitudinal dataset
+
+
+```python
+frame = sqlContext.sql("""
+SELECT client_id,
+    settings[1].locale AS locale,
+    map_keys(active_addons[1]) AS installed_addons,
+    geo_city[1] AS geoCity,
+    subsession_length[1] AS subsessionLength,
+    system_os[1].name AS os,
+    scalar_parent_browser_engagement_total_uri_count[1].value AS total_uri,
+    scalar_parent_browser_engagement_tab_open_event_count[1].value as tab_open_count,
+    places_bookmarks_count[1].sum as bookmark_count,
+    scalar_parent_browser_engagement_unique_domains_count[1].value as unique_tlds
+FROM longitudinal
+""")
+
+rdd = frame.rdd
+```
+
+## Loading addon data (AMO)
+
+We need to load the addon database to find out which addons are legacy addons.
+
+
+```python
+from taar.recommenders.utils import get_s3_json_content
+```
+
+
+```python
+AMO_DUMP_BUCKET = 'telemetry-parquet'
+AMO_DUMP_KEY = 'telemetry-ml/addon_recommender/addons_database.json'
+```
+
+
+```python
+amo_dump = get_s3_json_content(AMO_DUMP_BUCKET, AMO_DUMP_KEY)
+```
+
+## Filtering out legacy addons
+
+This is a helper function that takes a list of addon IDs and only returns the IDs that are from legacy addons.
+
+
+```python
+def get_legacy_addons(installed_addons):
+    legacy_addons = []
+    
+    for addon_id in installed_addons:
+        if addon_id in amo_dump:
+            addon = amo_dump[addon_id]
+            addon_files = addon.get('current_version', {}).get('files', {})
+
+            is_webextension = any([f.get("is_webextension", False) for f in addon_files])
+            is_legacy = not is_webextension
+
+            if is_legacy:
+                legacy_addons.append(addon_id)
+            
+    return legacy_addons
+```
+
+## Completing client data
+
+
+```python
+def complete_client_data(client_data):
+    client = client_data.asDict()
+    
+    client['installed_addons'] = client['installed_addons'] or []
+    client['disabled_addon_ids'] = get_legacy_addons(client['installed_addons'])
+    client['locale'] = str(client['locale'])
+    
+    return client
+```
+
+## Evaluating the existing recommenders
+
+To check if a recommender is able to make a recommendation, it's sometimes easier and cleaner to directly query it instead of checking the important attributes ourselves. For example, this is the case for the locale recommender.
+
+
+```python
+from taar.recommenders import CollaborativeRecommender, LegacyRecommender, LocaleRecommender
+```
+
+
+```python
+class DummySimilarityRecommender:
+    def can_recommend(self, client_data):
+        REQUIRED_FIELDS = ["geoCity", "subsessionLength", "locale", "os", "bookmark_count", "tab_open_count",
```

Comment:
nit: some whitespace issues here in the nptebook version.

Reply:
I guess this was because I copied that part from the taar repository. Should be fixed now.